tests/integration/chain_setup.go

```diff
@@ -104,16 +129,162 @@ func CreateAuctionBidMsg(t *testing.T, ctx context.Context, searcher cosmos.User
 
+	// set timeout height for generated txs
+	broadcaster.ConfigureFactoryOptions(func(txf tx.Factory) tx.Factory {
+		return txf.WithTimeoutHeight(timeoutHeight)
+	})
+
 	resp, err := cosmos.BroadcastTx(ctx, broadcaster, sender, msgs...)
+
+	// check execution was successful
+	require.NoError(t, err)
+	require.Equal(t, uint32(0), resp.Code)
+	return resp
+}
+
+// QueryBuilderParams queries the x/builder module's params
+func QueryBuilderParams(t *testing.T, chain ibc.Chain) buildertypes.Params {
+	// cast chain to cosmos-chain
+	cosmosChain, ok := chain.(*cosmos.CosmosChain)
+	require.True(t, ok)
+	// get nodes
+	nodes := cosmosChain.Nodes()
+	require.True(t, len(nodes) > 0)
+	// make params query to first node
+	resp, _, err := nodes[0].ExecQuery(context.Background(), "builder", "params")
+	require.NoError(t, err)
+
+	// unmarshal params
+	var params buildertypes.Params
+	err = json.Unmarshal(resp, &params)
 	require.NoError(t, err)
+	return params
+}
+
+// QueryAccountBalance queries a given account's balance on the chain
+func QueryAccountBalance(t *testing.T, chain ibc.Chain, address, denom string) int64 {
+	// cast the chain to a cosmos-chain
+	cosmosChain, ok := chain.(*cosmos.CosmosChain)
+	require.True(t, ok)
+	// get nodes
+	balance, err := cosmosChain.GetBalance(context.Background(), address, denom)
+	require.NoError(t, err)
+	return balance
+}
+
+// QueryAccountSequence
+func QueryAccountSequence(t *testing.T, chain *cosmos.CosmosChain, address string) uint64 {
+	// get nodes
+	nodes := chain.Nodes()
+	require.True(t, len(nodes) > 0)
+
+	resp, _, err := nodes[0].ExecQuery(context.Background(), "auth", "account", address)
+	require.NoError(t, err)
+	// unmarshal json response
+	var accResp codectypes.Any
+	require.NoError(t, json.Unmarshal(resp, &accResp))
+
+	// unmarshal into baseAccount
+	var acc authtypes.BaseAccount
+	require.NoError(t, acc.Unmarshal(accResp.Value))
+
+	return acc.GetSequence()
+}
+
+// Block returns the block at the given height
+func Block(t *testing.T, chain *cosmos.CosmosChain, height int64) *rpctypes.ResultBlock {
+	// get nodes
+	nodes := chain.Nodes()
+	require.True(t, len(nodes) > 0)
+
+	client := nodes[0].Client
+
+	resp, err := client.Block(context.Background(), &height)
+	require.NoError(t, err)
+
 	return resp
 }
+
+// WaitForHeight waits for the chain to reach the given height
+func WaitForHeight(t *testing.T, chain *cosmos.CosmosChain, height uint64) {
+	// wait for next height
+	err := testutil.WaitForCondition(30*time.Second, time.Second, func() (bool, error) {
+		pollHeight, err := chain.Height(context.Background())
+		if err != nil {
+			return false, err
+		}
+		return pollHeight > height, nil
+	})
+	require.NoError(t, err)
+}
+
+// VerifyBlock takes a Block and verifies that it contains the given bid at the 0-th index, and the bundled txs immediately after
+func VerifyBlock(t *testing.T, block *rpctypes.ResultBlock, bidTxHash string, txs [][]byte) {
```

Comment:
i would remove `bidTxHash` from func definition (this will help with testing vote extensions and lanes later on). I would add a skip index that lets u skip a few indices from the block.

Reply:
wym? This shld probs be private for now, and we can make public later as we need